### PR TITLE
Published content endpoint

### PR DESCRIPTION
**Pull request by michalstrzelecki**

### CODE REVIEW

File: src/Gzero/Cms/Http/Controllers/Api/PublicContentController.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?php namespace Gzero\Cms\Http\Controllers\Api;
+
+use Gzero\Cms\Http\Resources\ContentCollection;
+use Gzero\Cms\Repositories\ContentReadRepository;
+use Gzero\Core\Http\Controllers\ApiController;
+use Gzero\Core\Query\QueryBuilder;
+use Illuminate\Http\Request;
+
+/**
+ * Class DeletedContentController
```
Comment (AdrianSkierniewski) [inline src/Gzero/Cms/Http/Controllers/Api/PublicContentController.php]: You should update docs & swagger annotations

File: src/Gzero/Cms/Http/Controllers/Api/PublicContentController.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        $this->repository = $repository;
+        $this->request    = $request;
+    }
+
+    /**
+     * Display list of public contents
+     *
+     * @SWG\Get(
+     *   path="/public-contents",
+     *   tags={"content"},
```
Comment (AdrianSkierniewski) [inline src/Gzero/Cms/Http/Controllers/Api/PublicContentController.php]: we have public tag for those kind of endpoints

File: tests/functional/api/PublicContentCest.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+use Cms\FunctionalTester;
+use Gzero\Cms\Jobs\CreateContent;
+use Gzero\Core\Models\Language;
+use Gzero\Core\Models\User;
+
+class PublicContentCest {
+
+    public function _before(FunctionalTester $I)
+    {
+        $I->apiLoginAsAdmin();
```
Comment (AdrianSkierniewski) [inline tests/functional/api/PublicContentCest.php]: Why you're logging in as admin if this is public available endpoint?

File: tests/unit/repositories/ContentReadRepositoryTest.php [hunk 4]
```diff
@@ -223,5 +223,148 @@ public function canPaginateResults()
         $this->assertEquals('B title', $result->first()->translations->first()->title);
         $this->assertEquals('A title', $result->last()->translations->first()->title);
     }
+
+    /** @test */
+    public function getManyPublishedShouldHideTranslationsForInactiveRoutes()
```
Comment (AdrianSkierniewski) [inline tests/unit/repositories/ContentReadRepositoryTest.php]: I don't like those long title, could you remove method prefix and simplify them a bit?

File: routes/api.php [hunk 5]
```diff
@@ -21,6 +21,7 @@ function ($router) {
         $router->get('contents/{id}/children', 'NestedContentController@index');
 
         $router->get('deleted-contents', 'DeletedContentController@index');
+        $router->get('public-contents', 'PublicContentController@index');
```
Comment (AdrianSkierniewski) [inline routes/api.php]: This isn't right place for this route

File: tests/unit/repositories/ContentReadRepositoryTest.php [hunk 6]
```diff
@@ -223,5 +223,148 @@ public function canPaginateResults()
...
+        $this->assertCount(1, $content->routes);
+        $this->assertNotEmpty($content->routes->firstWhere('language_code', 'pl'));
+    }
+
+    /** @test */
+    public function getManyPublishedShouldReturnContentWitchHavePublishedAtDateInThePastAndDifferentThanNull()
+    {
+        $this->tester->haveContents([
+            [
+                'published_at' => Carbon::now()->subDays(1),
```
Comment (AdrianSkierniewski) [inline tests/unit/repositories/ContentReadRepositoryTest.php]: Why are you copying all those contents to see only if it works for null? I should be just a simple test that checks yet another case. We don't need that duplication

File: tests/unit/repositories/ContentReadRepositoryTest.php (hunk 4 above)
Comment (michalstrzelecki) [reply tests/unit/repositories/ContentReadRepositoryTest.php]: Yes, ofc.

**Pull request merged**